cleanlab/datalab/datalab.py

```diff
@@ -0,0 +1,682 @@
+# Copyright (C) 2017-2023  Cleanlab Inc.
+# This file is part of cleanlab.
+#
+# cleanlab is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cleanlab is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with cleanlab.  If not, see <https://www.gnu.org/licenses/>.
+"""
+Implements cleanlab's Datalab interface as a one-stop-shop for tracking
+and managing all kinds of issues in datasets.
+
+.. note::
+    .. include:: optional_dependencies.rst
+"""
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Union
+import warnings
+
+import numpy as np
+import numpy.typing as npt
+import pandas as pd
+
+import cleanlab
+from cleanlab.datalab.data import Data
+from cleanlab.datalab.data_issues import DataIssues
+from cleanlab.datalab.display import _Displayer
+from cleanlab.datalab.factory import _IssueManagerFactory, list_default_issue_types
+from cleanlab.datalab.serialize import _Serializer
+
+if TYPE_CHECKING:  # pragma: no cover
+    from datasets.arrow_dataset import Dataset
+    from scipy.sparse import csr_matrix
+
+    DatasetLike = Union[Dataset, pd.DataFrame, Dict[str, Any], List[Dict[str, Any]], str]
+
+__all__ = ["Datalab"]
+
+
+class Datalab:
+    """
+    A single object to automatically detect all kinds of issues in datasets.
+    This is how we recommend you interface with the cleanlab library if you want to audit the quality of your data. If you have other specific goals, then consider using the other methods across this library. Even then, Datalab may be the easiest way to run specific analyses of your data. Datalab tracks intermediate state (e.g. data statistics) from certain cleanlab functions that can be re-used across other cleanlab functions for better efficiency.
+
+    Parameters
+    ----------
+    data : Union[Dataset, pd.DataFrame, dict, list, str]
+        Dataset-like object that can be converted to a Hugging Face Dataset object.
+
+        It should contain the labels for all examples, identified by a
+        `label_name` column in the Dataset object.
+
+        See also
+        --------
+        :py:class:`Data <cleanlab.datalab.data.Data>`:
+        Internal class that represents the dataset.
```

Comment:
```suggestion
```

Reply:
recommend instead using this space to state the useful parts of documentation here directly. We try to avoid referring users to internal functions.